test/commands/ban.spec.ts

```diff
@@ -39,4 +78,115 @@ suite("ban command", () => {
     assert.equal(reasonOption.type, ApplicationCommandOptionType.String);
     assert.isTrue(reasonOption.required);
   });
+
+  test("should not ban self", async () => {
+    const interaction = new MockChatInputCommandInteraction({
+      commandName: "ban",
+      guild,
+      member,
+      user,
+      channel,
+      bot,
+    });
+    interaction.options.addOption({
+      name: "target",
+      value: user,
+      type: ApplicationCommandOptionType.User,
+    });
+    interaction.options.addOption({
+      name: "reason",
+      value: "test",
+      type: ApplicationCommandOptionType.String,
+    });
+    await ban.run({} as Camperbot, interaction.typeCast<"cached">());
+    assert.lengthOf(interaction.replies, 1);
+    assert.equal(interaction.replies[0].content, "You cannot ban yourself.");
+  });
+
+  test("should not ban bot", async () => {
+    const interaction = new MockChatInputCommandInteraction({
+      commandName: "ban",
+      guild,
+      member,
+      user,
+      channel,
+      bot,
+    });
+    interaction.options.addOption({
+      name: "target",
+      value: bot,
+      type: ApplicationCommandOptionType.User,
+    });
+    interaction.options.addOption({
+      name: "reason",
+      value: "test",
+      type: ApplicationCommandOptionType.String,
+    });
+    await ban.run(
+      { user: bot } as unknown as Camperbot,
+      interaction.typeCast<"cached">()
+    );
+    assert.lengthOf(interaction.replies, 1);
+    assert.equal(interaction.replies[0].content, "You cannot ban the bot.");
+  });
+
+  test("should not ban unbannable member", async () => {
+    const interaction = new MockChatInputCommandInteraction({
+      commandName: "ban",
+      guild,
+      member,
+      user,
+      channel,
+      bot,
+    });
+    interaction.options.addOption({
+      name: "target",
+      value: target,
+      type: ApplicationCommandOptionType.User,
+    });
+    interaction.options.addOption({
+      name: "reason",
+      value: "test",
+      type: ApplicationCommandOptionType.String,
+    });
+    await ban.run({} as Camperbot, interaction.typeCast<"cached">());
+    assert.lengthOf(interaction.replies, 1);
+    assert.equal(interaction.replies[0].content, "I cannot ban them.");
+  });
+
+  test("should ban a member", async () => {
+    const targetMember = guild.members.cache.get(target.id);
+    if (!targetMember) {
+      assert.fail("Target member not found");
+      return;
```

Comment:
Is this necessary? I suspect if you do an `assert.fail` it will immediately terminate the test.

If you're adding the `return` just to allow TS to infer `targetMember` is set, then consider just using optional chaining or a null-safe assertion (e.g. `targetMember!.bannable = true`).